jetty-core/jetty-fcgi/jetty-fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/internal/ServerFCGIConnection.java

```diff
@@ -214,17 +225,26 @@ void parseAndFill()
     {
         if (LOG.isDebugEnabled())
             LOG.debug("parseAndFill {}", this);
+        acquireInputBuffer();
         // This loop must run only until the request is completed.
         // See also HttpConnection.parseAndFillForContent().
         while (stream != null)
         {
+            // The networkBuffer cannot be released immediately after parse()
+            // even if the buffer has been fully consumed because releaseInputBuffer()
+            // must be called as the last release for it to be able to null out the
+            // networkBuffer field exactly when the latter isn't used anymore.
             if (parse(networkBuffer.getByteBuffer()))
-                return;
+                break;
```

Comment:
again, no difference between `break` and `return` here, so I kind of prefer `return` as I don't need to scroll down to see what happens next

Reply:
Same as above: for style, so I'll change `break` for `return`.